heim-host/src/sys/windows/users.rs

```diff
@@ -27,6 +31,44 @@ impl User {
         }))
     }
 
+    pub fn try_from_sid(sid: PSID) -> Result<Self> {
+        // name and domain cannot be longer than 256
+        let mut name_cch: DWORD = 256;
+        let mut name: Vec<WCHAR> = Vec::with_capacity(name_cch as usize);
+        let mut domain_cch: DWORD = 256;
+        let mut domain: Vec<WCHAR> = Vec::with_capacity(domain_cch as usize);
+
+        // winapi does not have SID_NAME_USE enum
+        let mut account_type: SID_NAME_USE = 0;
+
+        let result = unsafe {
+            LookupAccountSidW(
+                ptr::null(),
+                sid,
+                name.as_mut_ptr(),
+                &mut name_cch,
+                domain.as_mut_ptr(),
+                &mut domain_cch,
+                &mut account_type as PSID_NAME_USE,
+            )
+        };
+
+        if result == 0 {
+            return Err(Error::last_os_error());
+        }
+
+        unsafe {
+            name.set_len(name_cch as usize);
+            domain.set_len(domain_cch as usize);
+        }
+
+        Ok(Self {
+            domain: String::from_utf16(domain.as_slice()).unwrap(),
+            username: String::from_utf16(name.as_slice()).unwrap(),
```

Comment:
I'm not sure if it possible to set such a value for username or domain that it will be impossible to convert it into the utf-8, so error propagation seems okay for now.
Instead of the unwraps, `impl From<std::string::FromUtf16Error> for Error` should be added [in here](
https://github.com/heim-rs/heim/blob/37839cfe9658feb98fcd0156a11a77dfb028b3cb/heim-common/src/errors.rs#L322), so we can use `?` instead.